frontends/ol-ckeditor/src/components/CkeditorArticle.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useMemo, useEffect } from "react"
+import { CKEditor } from "@ckeditor/ckeditor5-react"
+
+import { ClassicEditor } from "@ckeditor/ckeditor5-editor-classic"
+import type { EditorConfig } from "@ckeditor/ckeditor5-core"
+
+import { Essentials } from "@ckeditor/ckeditor5-essentials"
+import { UploadAdapter } from "@ckeditor/ckeditor5-adapter-ckfinder"
+import { Autoformat } from "@ckeditor/ckeditor5-autoformat"
+import { Bold, Italic } from "@ckeditor/ckeditor5-basic-styles"
+import { BlockQuote } from "@ckeditor/ckeditor5-block-quote"
+import { EasyImage } from "@ckeditor/ckeditor5-easy-image"
+import { Heading } from "@ckeditor/ckeditor5-heading"
+import {
+  Image,
+  ImageStyle,
+  ImageToolbar,
+  ImageUpload
+} from "@ckeditor/ckeditor5-image"
+import { Link } from "@ckeditor/ckeditor5-link"
+import { List } from "@ckeditor/ckeditor5-list"
+import { MediaEmbed } from "@ckeditor/ckeditor5-media-embed"
+import { Paragraph } from "@ckeditor/ckeditor5-paragraph"
+import { CloudServices } from "@ckeditor/ckeditor5-cloud-services"
+
+// block toolbar setup
+import { BlockToolbar } from "@ckeditor/ckeditor5-ui"
+import { ParagraphButtonUI } from "@ckeditor/ckeditor5-paragraph"
+import getCloudServicesConfig from "./cloudServices"
+import { ensureEmbedlyPlatform, embedlyCardHtml } from "ol-util"
+
+const baseEditorConfig: EditorConfig = {
+  plugins: [
+    Essentials,
+    Autoformat,
+    Bold,
+    Italic,
+    BlockQuote,
+    Heading,
+    Link,
+    List,
+    MediaEmbed,
+    Paragraph,
+    BlockToolbar,
+    UploadAdapter,
+    CloudServices,
+    EasyImage,
+    Image,
+    ImageStyle,
+    ImageToolbar,
+    ImageUpload,
```

Comment:
You might notice an error in the console:
> index.js:5 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'model')

(A very similar `console.error` appears in OCW Studio.)

As far as I can tell, this has no noticeable consequences other than the console error, and it seems to be a bug in `ckeditor5-react` + ImageUpload plugin; see https://github.com/ckeditor/ckeditor5-react/issues/370